core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -450,10 +450,16 @@ protected DeterministicKeyChain(KeyCrypter crypter, AesKey aesKey, Deterministic
         }
     }
 
-    public HDPath getAccountPath() {
+    public HDPath.HDPartialPath getAccountPath() {
         return accountPath;
     }
 
+    public HDPath.HDFullPath accountFullPath() {
+        boolean hasPrivateKey = !getWatchingKey().isWatching();
+        return accountPath.asFull(hasPrivateKey ? HDPath.Prefix.PRIVATE : HDPath.Prefix.PUBLIC);
+    }
+
+
```

Comment:
I think this could be extracted from the first commit and merged with the usages in tests (basically the other 4 commits).

There is a surplus newline here.

Reply:
Resolved in new PR #3765